osu.Game/Online/Rooms/PlaylistExtensions.cs

```diff
@@ -39,6 +41,17 @@ public static IEnumerable<PlaylistItem> GetUpcomingItems(this IEnumerable<Playli
         }
 
         public static string GetTotalDuration(this BindableList<PlaylistItem> playlist) =>
-            playlist.Select(p => p.Beatmap.Length).Sum().Milliseconds().Humanize(minUnit: TimeUnit.Second, maxUnit: TimeUnit.Hour, precision: 2);
+            playlist.Select(p =>
+            {
+                var ruleset = p.Beatmap.Ruleset.CreateInstance();
```

Comment:
Instantiation of rulesets is computationally expensive because of the underlying implementation which relies on reflection. So the ruleset should be instantiated only when strictly necessary, namely when there are required mods present.